rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -349,4 +382,24 @@ private boolean isTargetRecursivePackageName(String packageName) {
         }
 
     }
+
+    @Value
+    @EqualsAndHashCode(callSuper = false)
+    private static class TypeReferenceChangePackageVisitor extends TreeVisitor<Tree, ExecutionContext> {
+        Set<TypeReference> matches;
+        String oldPackageName;
+        String newPackageName;
+
+        @Override
+        public @Nullable Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
+            Tree tree1 = super.visit(tree, ctx);
+            for (TypeReference ref : matches) {
+                if (ref.getTree().equals(tree) && ref.supportsRename()) {
+                    return ref.renameTo(ref.getName().replace(oldPackageName, newPackageName)).visit(tree, ctx);
```

Comment:
Not sure if we should rely on replacing the entire reference or whether we actually need a `renamePackageTo` method on the `TypeReference` interface